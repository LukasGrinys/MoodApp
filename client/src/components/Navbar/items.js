import { faHome, faSignInAlt, faSignOutAlt, faUserPlus, faCog, faBook } from '@fortawesome/free-solid-svg-icons';

export const items = [
    {
        iconName: faHome,
        path: "/",
        text: "Home",
        isAuth: null
    },
    {
        iconName: faSignInAlt,
        path: "/login",
        text: "Log In",
        isAuth: false
    },
    {
        iconName: faSignOutAlt,
        path: "/logout",
        text: "Log Out",
        isAuth: true
    },
    {
        iconName: faUserPlus,
        path: "/register",
        text: "Sign Up",
        isAuth: null
    },
    {
        iconName: faBook,
        path: "/dashboard",
        text: "Your Dashboard",
        isAuth: true
    },
    {
        iconName: faCog,
        path: "/settings",
        text: "Settings",
        isAuth: true
    }
];