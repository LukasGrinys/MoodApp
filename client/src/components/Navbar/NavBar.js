import React from 'react';
import { Link } from 'react-router-dom';
import styles from './NavBar.module.scss';
import { items } from './items';
import { useTheme } from '../../contexts/ThemeContext';
import FontAwesome from 'react-fontawesome';
import classNames from 'classnames';

const NavBar = ({isAuth, toggleNav}) => {
    const darkTheme = useTheme();

    const itemElement = (item, i) => (
        <Link to={item.path} key={i}>
            <div className={styles.navItem} onClick={toggleNav}>
                <FontAwesome name={item.iconName}/> {item.text}
            </div> 
        </Link>
    )

    const renderItems = (items) => {
        return (
            items.map( (item, i) => {
                if (isAuth && item.isAuth) {
                    return itemElement(item,i)     
                } else if (!isAuth && item.isAuth === false) {
                    return itemElement(item,i) 
                } else if (item.isAuth === null) {
                    return itemElement(item,i)
                }
                return null
            })
        )
    }

    return (
        <div>
            <div className={classNames(styles.navBar, darkTheme && styles.nightMode)}>
                {renderItems(items)}
            </div>
            <div className={styles.navBarMask} onClick={toggleNav}></div>
        </div> 
    );
};

export default NavBar;