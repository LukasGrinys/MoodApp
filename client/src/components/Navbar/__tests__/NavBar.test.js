import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import NavBar from '../NavBar';
import { items as navbarItems } from '../items';
import { mount } from 'enzyme';

const toggleNavMock = jest.fn();

const navbarProps = {
    isAuth: true,
    toggleNav: toggleNavMock
};

const wrapper = mount(
    <Router>
        <NavBar {...navbarProps}/>
    </Router>
);

test('Renders properly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('It has navigation items', () => {
    const authItems = navbarItems.filter( item => item.isAuth !== false );

    expect(wrapper.find('.navItem').length).toBe(authItems.length);
});

test('It has links to item', () => {
    const firstItem = navbarItems.find( item => item.isAuth !== false );
    const firstPath = firstItem.path;

    expect(wrapper.find('Link').at(0).prop('to')).toBe(firstPath);
});

test('It has a mask, covering page content', () => {
    expect(wrapper.find('.navBarMask').length).toBe(1);
});

test('It closes nav when clicked on mask', () => {
    wrapper.find('.navBarMask').simulate('click');

    expect(toggleNavMock).toHaveBeenCalledTimes(1);
});