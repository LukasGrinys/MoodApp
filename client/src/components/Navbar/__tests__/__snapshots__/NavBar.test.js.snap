// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders properly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NavBar
      isAuth={true}
      toggleNav={[MockFunction]}
    >
      <div>
        <div
          className="navBar"
        >
          <Link
            key="0"
            to="/"
          >
            <LinkAnchor
              href="/"
              navigate={[Function]}
            >
              <a
                href="/"
                onClick={[Function]}
              >
                <div
                  className="navItem"
                  onClick={[MockFunction]}
                >
                  <FontAwesome
                    name="home"
                  >
                    <span
                      aria-hidden={true}
                      className="fa fa-home"
                    />
                  </FontAwesome>
                   
                  Home
                </div>
              </a>
            </LinkAnchor>
          </Link>
          <Link
            key="2"
            to="/logout"
          >
            <LinkAnchor
              href="/logout"
              navigate={[Function]}
            >
              <a
                href="/logout"
                onClick={[Function]}
              >
                <div
                  className="navItem"
                  onClick={[MockFunction]}
                >
                  <FontAwesome
                    name="sign-out"
                  >
                    <span
                      aria-hidden={true}
                      className="fa fa-sign-out"
                    />
                  </FontAwesome>
                   
                  Log Out
                </div>
              </a>
            </LinkAnchor>
          </Link>
          <Link
            key="3"
            to="/register"
          >
            <LinkAnchor
              href="/register"
              navigate={[Function]}
            >
              <a
                href="/register"
                onClick={[Function]}
              >
                <div
                  className="navItem"
                  onClick={[MockFunction]}
                >
                  <FontAwesome
                    name="user-plus"
                  >
                    <span
                      aria-hidden={true}
                      className="fa fa-user-plus"
                    />
                  </FontAwesome>
                   
                  Sign Up
                </div>
              </a>
            </LinkAnchor>
          </Link>
          <Link
            key="4"
            to="/dashboard"
          >
            <LinkAnchor
              href="/dashboard"
              navigate={[Function]}
            >
              <a
                href="/dashboard"
                onClick={[Function]}
              >
                <div
                  className="navItem"
                  onClick={[MockFunction]}
                >
                  <FontAwesome
                    name="book"
                  >
                    <span
                      aria-hidden={true}
                      className="fa fa-book"
                    />
                  </FontAwesome>
                   
                  Your Dashboard
                </div>
              </a>
            </LinkAnchor>
          </Link>
          <Link
            key="5"
            to="/settings"
          >
            <LinkAnchor
              href="/settings"
              navigate={[Function]}
            >
              <a
                href="/settings"
                onClick={[Function]}
              >
                <div
                  className="navItem"
                  onClick={[MockFunction]}
                >
                  <FontAwesome
                    name="cog"
                  >
                    <span
                      aria-hidden={true}
                      className="fa fa-cog"
                    />
                  </FontAwesome>
                   
                  Settings
                </div>
              </a>
            </LinkAnchor>
          </Link>
        </div>
        <div
          className="navBarMask"
          onClick={[MockFunction]}
        />
      </div>
    </NavBar>
  </Router>
</BrowserRouter>
`;
